### DISCUSSION TITLE
On changing an edge place from a Node to the other, it can create a new unwanted node

### DISCUSSION THREAD

**Discussion by zakaria-imzilen** [xyflow/xyflow#4151] [Q&A] [answered]
While using this feature: https://reactflow.dev/examples/nodes/add-node-on-edge-drop
that allows the creation of new Nodes on edge drop, I noticed that it sometimes gets triggered even when changing an edge target handle.

Example: while having an already edge between two nodes and I'd like to change its target handle from one to another, while dragging the edge, if I don't put it right in the desired handle, a new unwanted node will be created.

Desired Behavior: If I miss dropping the edge on the correct handle, the edge should point out to the old handle.

Do you have any solution, please?

**Comment by bcakmakoglu** [comment, marked as answer]
How about holding a flag during an edge update and not allowing a new node to be generated if it's during an edge update?

```ts
const [isUpdating, setIsUpdating] = useState(false)

const onConnectEnd = useCallback(
    (event) => {
      if (!connectingNodeId.current || isUpdating) return;
	  // ... the rest of the code
	}, [/* other deps */, isUpdating]
)

const onEdgeUpdateState = () => {
	setIsUpdating(true)
}

const onEdgeUpdateEnd = () => {
	setIsUpdating(false)
}
```

**Comment by zakaria-imzilen** [comment]
Yees it works, thank you.
